packages/adapter-mariadb/src/mariadb.ts

```diff
@@ -170,19 +172,38 @@ export class PrismaMariaDbAdapterFactory implements SqlDriverAdapterFactory {
 
   #capabilities?: Capabilities
   #config: mariadb.PoolConfig | string
+  #externalPool: mariadb.Pool | null
```

Comment:
This needs to be a tagged union, we either have a config or a pool (since we can't retrieve a config from a pool with `mariadb` driver). We should also check that `connect` can't be called more than once if an external pool is used and `disposeExternalPool` is true.